include/deal.II/lac/precondition.h

```diff
@@ -2819,7 +2819,9 @@ namespace internal
             //           + f_2 * P^{-1} * (b-A*x^{n})
             DEAL_II_OPENMP_SIMD_PRAGMA
             for (std::size_t i = begin; i < end; ++i)
-              solution_old[i] =
+              // for efficiency reason, write back to temp_vector that is
+              // already read (avoid read-for-ownership)
```

Comment:
I think this comment is wrong. It is not the read that gets avoid (as we read `solution_old` below), but we can avoid writing `temp_vector` back to memory by replacing its volatile content (only needed in the subsequent formulas) by over-writing it with the useful content that previously went into `solution_old`.

Reply:
I have updated the comment.